app/page/authorization/transaction.tsx

```diff
@@ -131,161 +127,78 @@ function UserOperationConfirmation(props: {
-        tx.networkId,
-        await sender.getEntryPoint()
-      )
-      const execution = await sender.buildExecution(tx)
-      const userOp =
-        transactionType === TransactionType.ERC4337V0_6
-          ? UserOperationV0_6.wrap(execution)
-          : UserOperationV0_7.wrap(execution)
-      await estimateGas(userOp)
-      setUserOp(userOp)
-    }
-    setupUserOp()
-  }, [tx.id])
-
-  useEffect(() => {
-    async function estimateUserOp() {
-      setUserOpEstimating(true)
-      await estimateGas(userOp)
-      setUserOp(userOp)
-      setUserOpEstimating(false)
-    }
-    if (!userOp) {
-      return
-    }
-    estimateUserOp()
-  }, [paymentOption])
+  const {
+    userOp,
+    userOpResolving,
+    sendUserOperation,
+    rejectUserOperation,
+    userOpEstimating
+  } = useUserOperation({ ...props, paymentOption })
 
-  useEffect(() => {
-    async function calculatePayment() {
-      setPaymentCalculating(true)
-      setPayment({
-        ...payment,
-        tokenFee: await paymentOption.paymaster.quoteFee(
-          userOp.calculateGasFee(),
-          ETH
-        )
-      })
-      setPaymentCalculating(false)
-    }
-    if (!userOp) {
-      return
-    }
-    calculatePayment()
-  }, [JSON.stringify(userOp?.unwrap())])
+  const { payment, paymentCalculating } = usePayment({ userOp, paymentOption })
 
   if (!userOp) {
     return
   }
-
   return (
-    <div>
-      <div>
-        <h1>Transaction Detail</h1>
-        <div>
-          {Object.entries(userOp.unwrap()).map(([key, value], i) => {
-            return (
-              <div key={i}>
-                {key}: {`${value}`}
-              </div>
-            )
-          })}
+    <>
+      <Header title={"Send"} href={Path.Index}>
+        <div className="text-[48px]">1.2 ETH</div>
+      </Header>
+      <section className="py-[16px] text-[16px]">
+        <h2 className="py-[12px]">From</h2>
+        <div className="flex gap-[12px] items-center">
+          <Wallet />
+          <div className="w-[322px] py-[9.5px]">
+            <h3>Jesse's wallet</h3>
+            <h4 className="text-[#989898] ">{userOp.sender}</h4>
+          </div>
         </div>
+        <h2 className="py-[12px]">To</h2>
+        <div className="flex gap-[12px] items-center">
+          <Wallet />
+          <div className="py-[16px] w-[322px]">
+            <h3>{props.tx.to}</h3>
+          </div>
+        </div>
+      </section>
+      <div className="relative w-full">
+        <hr className="divider" />
+      </div>
+      <section className="py-[16px] text-[16px]">
+        <h2 className="py-[8px]">Est. gas fee</h2>
+        <div className="flex gap-[12px] py-[16px]">
+          <Gas />
+          <p>
+            {userOpEstimating || paymentCalculating
+              ? "Estimating..."
+              : `${ethers.formatEther(
+                  userOp ? userOp.calculateGasFee() : 0n
+                )} ${ETH.symbol}`}
+          </p>
+          <p>= ~$ ? USD</p>
```

Comment:
We haven't yet integrated any price oracle, I think we can omit price conversion for now.

Reply:
[Remove unused code](https://github.com/pilagod/waallet-extension/pull/152/commits/1b494b04552b635cbcb966450d6b90cc21c49294)